cf-app/src/main/kotlin/space/crowdforce/configuration/TelegramConfig.kt

```diff
@@ -1,6 +1,37 @@
 package space.crowdforce.configuration
 
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.event.ApplicationReadyEvent
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.event.EventListener
+import org.telegram.telegrambots.ApiContextInitializer
+import org.telegram.telegrambots.meta.TelegramBotsApi
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException
+import space.crowdforce.service.tg.CustomTelegramBot
 
 @Configuration
-class TelegramConfig
+class TelegramConfig {
+
+    @Autowired
+    lateinit var crowdforce: CustomTelegramBot;
+
+    @EventListener(ApplicationReadyEvent::class)
+    fun botInit() {
+        ApiContextInitializer.init();
+
+        val botsApi = TelegramBotsApi()
+
+        try {
+            botsApi.registerBot(crowdforce)
```

Comment:
it's maybe an issue because the bot will be registered only after the spring context is initialized. but other beans which dependent on the bot can access API methods during context initialization. I'd create a telegram initializer bean (TelegramBotsApi) and move the registerBot method to the CustomTelegramBot's post construct method.

Reply:
ok